src/Cell.cpp

```diff
@@ -1,70 +1,64 @@
 
-
 #include "TMath.h"
 #ifdef STANDALONE
 #include "Cell.h"
 #else
 #include "HGCalSimulation/FastShower/interface/Cell.h"
 #endif
 
+#include <vector>
+
+
+uint32_t Cell::id(uint16_t i, uint16_t j, uint16_t k) {
+    // id format = iiijjjkk
+    return i*100000 + j*100 + k;
+}
+
 
-uint32_t Cell::id(uint16_t i, uint16_t j)
+Cell::Cell(TVectorD&& position,
+           std::vector<TVectorD>&& vertices,
+           double orientation,
+           int i_index,
+           int j_index,
+           int k_index):
+            position_(std::move(position)),
+            vertices_(std::move(vertices)),
+            orientation_(orientation)
 {
-  uint32_t id = 0;
-  id |= i;
-  id |= (j<<16);
-  return id;
+
+  i_index_ = (int16_t)i_index;
+  j_index_ = (int16_t)j_index;
+  k_index_ = (int16_t)k_index;
+  id_ = Cell::id(i_index_, j_index_, k_index_);
+  x_ = this->getPosition()(0);
```

Comment:
Why a duplication of the position into `x` `y` `z` variables?

Reply:
Fixed